diff --git a/lib/lib/index.js b/lib/lib/index.js
index a00ee05c0d3f99ae494c41b4d9c1267bccd79687..7f4b6fb230d6bec6095a6ef333622fc55781de75 100644
--- a/lib/lib/index.js
+++ b/lib/lib/index.js
@@ -8,6 +8,9 @@ var _runWorkboxBuild = _interopRequireDefault(require("./run-workbox-build"));
 var _injectSwRegister = _interopRequireDefault(require("./inject-sw-register"));
 function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
 async function _default() {
+  if (process.env.HEXO_OFFLINE_SKIP === "1") {
+    return Promise.resolve();
+  }
   await _runWorkboxBuild.default.call(this);
   return (0, _injectSwRegister.default)(this.public_dir);
 }
\ No newline at end of file
diff --git a/lib/lib/run-workbox-build.js b/lib/lib/run-workbox-build.js
index f896ae8d76e23a8aa55dccbdd931dc283f71bc73..8ff29644c750488c31c71b53335d442ff6d13e20 100644
--- a/lib/lib/run-workbox-build.js
+++ b/lib/lib/run-workbox-build.js
@@ -6,7 +6,6 @@ Object.defineProperty(exports, "__esModule", {
 exports.default = runWorkboxBuild;
 var _fs = _interopRequireDefault(require("fs"));
 var _path = _interopRequireDefault(require("path"));
-var _workboxBuild = require("workbox-build");
 var _constants = require("./constants");
 function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
 function generateSWPrecacheConfig({
@@ -50,6 +49,7 @@ async function runWorkboxBuild() {
   if (!_fs.default.existsSync(indexHTMLPath)) {
     return Promise.resolve();
   }
+  const _workboxBuild = require("workbox-build");
   let offline;
   if ("offline" in config) {
     logger.warn(`
